### DISCUSSION TITLE
Extracting Data From Project Board

### DISCUSSION THREAD

**Discussion by RicardoDL89** [community/community#52140] [Projects and Issues]
### Select Topic Area

Question

### Body

Hi eveyone

Does anyone know how to extract the data from the project board correctly?

I have tried using the api however I keep getting the error 'Error: Unable to fetch data from GitHub API (Status code: 404)",

I am not sure the PAT I am using is the correct way of doing this,

If anyone can guide me on this I will really appreciate the assistance

**Comment by AneeqMalik** [comment]
To extract data from a project board in GitHub, you can use the GitHub REST API. Here are the general steps you can follow:

Generate a personal access token (PAT): In order to access the GitHub API, you will need to generate a PAT with the appropriate permissions. Go to your GitHub account settings, click on "Developer settings", and then click on "Personal access tokens". Click on "Generate new token", give your token a name, and select the appropriate scopes. For accessing project boards, you will need the "repo" scope.

Make a GET request to the project board API endpoint: To get the data from a project board, you will need to make a GET request to the API endpoint for the project board. The URL for the API endpoint should look something like this: https://api.github.com/projects/:project_id. Replace ":project_id" with the ID of the project board you want to access.

Include the PAT in the request headers: In order to authenticate your API request, you will need to include your PAT in the request headers. You can do this by adding an "Authorization" header with the value "Bearer <your_pat>". Replace "<your_pat>" with the actual value of your PAT.

Here's an example of how you can make a GET request to the project board API endpoint using Node.js:

```
const axios = require('axios');

const projectBoardId = '<your_project_board_id>';
const pat = '<your_personal_access_token>';

const options = {
  headers: {
    'Authorization': `Bearer ${pat}`,
    'Accept': 'application/vnd.github.inertia-preview+json' // This is required for accessing project boards
  }
};

const url = `https://api.github.com/projects/${projectBoardId}`;

axios.get(url, options)
  .then(response => {
    console.log(response.data);
  })
  .catch(error => {
    console.log(error);
  });

```
Note that you will need to replace "<your_project_board_id>" and "<your_personal_access_token>" with the actual values of your project board ID and PAT, respectively.

Also, make sure that the PAT you are using has the necessary permissions to access the project board. If you are still getting a 404 error, double-check that you are using the correct project board ID in your API request.

**Comment by RicardoDL89** [comment]
Thank you I keep getting the 401 Unauthorized code. I have checked my PAT has the neccessary permissions, I am also an admin in the repo where the project lives. The project is under an organization, will this cause the issue with the authoriazation? Even the repo owner's PAT is giving the same 401 code
